### PR TITLE
[Refactoring] Removed custom exif rotation method

### PR DESCRIPTION
**Pull request by bsekachev**
### Motivation and context
* We do not need custom exif rotation method, since PIL already has one.
* Added some closings for PIL Images

> The lifecycle of a single-frame image is relatively simple. The file must remain open until the load() or close() function is called or the context manager exits.

https://pillow.readthedocs.io/en/stable/reference/open_files.html#image-lifecycle

Means, that if we opened file and load() was not called internally, the file will stay opened. So, I added context manager and called close() in somewhere just in case

### Checklist

- [x] I submit my changes into the `develop` branch
- [ ] I have added a description of my changes into the [CHANGELOG](https://github.com/opencv/cvat/blob/develop/CHANGELOG.md) file
- [ ] I have updated the documentation accordingly
- [ ] I have added tests to cover my changes
- [ ] I have linked related issues (see [GitHub docs](
  https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword))
- [ ] I have increased versions of npm packages if it is necessary
  ([cvat-canvas](https://github.com/opencv/cvat/tree/develop/cvat-canvas#versioning),
  [cvat-core](https://github.com/opencv/cvat/tree/develop/cvat-core#versioning),
  [cvat-data](https://github.com/opencv/cvat/tree/develop/cvat-data#versioning) and
  [cvat-ui](https://github.com/opencv/cvat/tree/develop/cvat-ui#versioning))

### License

- [x] I submit _my code changes_ under the same [MIT License](
  https://github.com/opencv/cvat/blob/develop/LICENSE) that covers the project.
  Feel free to contact the maintainers if that's a concern.

### CODE REVIEW

File: cvat/apps/engine/frame_provider.py [hunk 1]
```diff
@@ -175,7 +174,7 @@ def get_preview(self, frame_number):
         else:
             preview, _ = self.get_frame(frame_number, self.Quality.COMPRESSED, self.Type.PIL)
 
-        preview = rotate_within_exif(preview)
+        preview = ImageOps.exif_transpose(preview)
```
Comment (nmanovic) [inline cvat/apps/engine/frame_provider.py:177]:
```suggestion
        ImageOps.exif_transpose(preview, in_place=True)
```

File: cvat/apps/engine/media_extractors.py [hunk 2]
```diff
@@ -127,7 +111,7 @@ def _get_preview(obj):
             preview = Image.open(obj)
         else:
             preview = obj
-        preview = rotate_within_exif(preview)
+        preview = ImageOps.exif_transpose(preview)
```
Comment (nmanovic) [inline cvat/apps/engine/media_extractors.py:114]:
```suggestion
        ImageOps.exif_transpose(preview, in_place=True)
```

File: cvat/apps/engine/media_extractors.py [hunk 3]
```diff
@@ -604,8 +588,12 @@ def __init__(self, quality, dimension=DimensionType.DIM_2D):
 
     @staticmethod
     def _compress_image(image_path, quality):
-        image = image_path.to_image() if isinstance(image_path, av.VideoFrame) else Image.open(image_path)
-        image = rotate_within_exif(image)
+        if isinstance(image_path, av.VideoFrame):
+            image = image_path.to_image()
+        else:
+            with Image.open(image_path) as source_image:
+                image = ImageOps.exif_transpose(source_image)
```
Comment (nmanovic) [inline cvat/apps/engine/media_extractors.py:595]:
```suggestion
            image = Image.open(image_path)
            ImageOps.exif_transpose(image, in_place=True)
```

File: cvat/apps/engine/media_extractors.py [hunk 4]
```diff
@@ -631,13 +619,14 @@ def _compress_image(image_path, quality):
             image = ImageOps.equalize(image)         # The Images need equalization. High resolution with 16-bit but only small range that actually contains information
 
         converted_image = image.convert('RGB')
-        image.close()
```
Comment (nmanovic) [inline cvat/apps/engine/media_extractors.py:634]: Why did you remove image.close()?

File: cvat/apps/engine/frame_provider.py (hunk 1 above)
Comment (bsekachev) [reply cvat/apps/engine/frame_provider.py:177]: in_place is not supported in Pillow 9.3.0.
It was implemented starting Pillow 10
